src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,169 @@
+                    "fixed length parameter [" + OVERLAP_RATE_FIELD + "] must be between 0 and " + OVERLAP_RATE_UPPER_BOUND
+                );
+            }
+        }
+
+        if (parameters.containsKey(TOKENIZER_FIELD) && !(parameters.get(TOKENIZER_FIELD) instanceof String)) {
+            throw new IllegalArgumentException(
+                "fixed length parameter [" + TOKENIZER_FIELD + "] cannot be cast to [" + String.class.getName() + "]"
+            );
+        }
+    }
+
+    /**
+     * Return the chunked passages for fixed token length algorithm
+     *
+     * @param content input string
+     * @param parameters a map containing parameters, containing the following parameters
+     * 1. tokenizer the analyzer tokenizer in opensearch, please check https://opensearch.org/docs/latest/analyzers/tokenizers/index/
+     * 2. token_limit the token limit for each chunked passage
+     * 3. overlap_rate the overlapping degree for each chunked passage, indicating how many token comes from the previous passage
+     * 4. max_token_count the max token limit for the tokenizer
+     */
+    @Override
+    public List<String> chunk(String content, Map<String, Object> parameters) {
+        // prior to chunking, parameters have been validated
+        int tokenLimit = DEFAULT_TOKEN_LIMIT;
+        BigDecimal overlap_rate = new BigDecimal(String.valueOf(DEFAULT_OVERLAP_RATE));
+        int maxTokenCount = DEFAULT_MAX_TOKEN_COUNT;
+
+        String tokenizer = DEFAULT_TOKENIZER;
+
+        if (parameters.containsKey(TOKEN_LIMIT_FIELD)) {
+            tokenLimit = ((Number) parameters.get(TOKEN_LIMIT_FIELD)).intValue();
+        }
+        if (parameters.containsKey(OVERLAP_RATE_FIELD)) {
+            overlap_rate = new BigDecimal(String.valueOf(parameters.get(OVERLAP_RATE_FIELD)));
+        }
+        if (parameters.containsKey(MAX_TOKEN_COUNT_FIELD)) {
+            maxTokenCount = ((Number) parameters.get(MAX_TOKEN_COUNT_FIELD)).intValue();
+        }
+        if (parameters.containsKey(TOKENIZER_FIELD)) {
+            tokenizer = (String) parameters.get(TOKENIZER_FIELD);
+        }
+
+        List<String> tokens = tokenize(content, tokenizer, maxTokenCount);
+        List<String> passages = new ArrayList<>();
+
+        String passage;
+        int startToken = 0;
+        BigDecimal overlapTokenNumberBigDecimal = overlap_rate.multiply(new BigDecimal(String.valueOf(tokenLimit)))
+            .setScale(0, RoundingMode.DOWN);
+        int overlapTokenNumber = overlapTokenNumberBigDecimal.intValue();
+        ;
+        // overlapTokenNumber must be smaller than the token limit
+        overlapTokenNumber = Math.min(overlapTokenNumber, tokenLimit - 1);
+
+        while (startToken < tokens.size()) {
+            if (startToken + tokenLimit >= tokens.size()) {
+                // break the loop when already cover the last token
+                passage = String.join(" ", tokens.subList(startToken, tokens.size()));
```

Comment:
It seems " " is for default tokenizer, what if user used a different tokenizer, should we still use " " here?

Reply:
Good point. " " is not applicable to all tokenizers. The tokens returned by the analyzer tokenizer contains offset information: `start_offset` and `end_offset`. I will use these two offsets to take the substring from the input `content`.

Reply:
Already changed. Please help review.